### PR TITLE
WINDUP-528:  Quiet down the logging on unparseable XML files

### PR DESCRIPTION
**Pull request by jsight**
https://issues.jboss.org/browse/WINDUP-528

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/686/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/56/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/687/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/57/
Test PASSed.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/DiscoverMavenProjectsRuleProvider.java [hunk 1]
```diff
@@ -163,10 +164,15 @@ private void addFilesToModel(MavenProjectModel mavenProjectModel, FileModel file
...
     }
 
-    public MavenProjectModel extractMavenProjectModel(GraphRewrite event, String defaultProjectName, XmlFileModel xmlResourceModel)
+    public MavenProjectModel extractMavenProjectModel(GraphRewrite event, String defaultProjectName, XmlFileModel xmlFileModel)
     {
-        File myFile = xmlResourceModel.asFile();
-        Document document = xmlResourceModel.asDocument();
+        File xmlFile = xmlFileModel.asFile();
+        Document document = new XmlFileService(event.getGraphContext()).loadDocumentQuiet(xmlFileModel);
+        if (document == null)
```
Comment (lincolnthree) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/DiscoverMavenProjectsRuleProvider.java]: Good.

File: rules-xml/addon/src/main/java/org/jboss/windup/rules/apps/xml/DiscoverXmlFilesRuleProvider.java [hunk 2]
```diff
@@ -128,8 +131,15 @@ private void addXmlMetaInformation(GraphContext context, FileModel file)
         }
         catch (SAXException e)
         {
-            LOG.log(Level.WARNING, "Failed to parse xml entity: " + file.getFilePath() + ", due to: " + e.getMessage(),
+            if (file.asFile().length() == 0)
```
Comment (lincolnthree) [inline rules-xml/addon/src/main/java/org/jboss/windup/rules/apps/xml/DiscoverXmlFilesRuleProvider.java]: Good.

File: rules-xml/addon/src/main/java/org/jboss/windup/rules/apps/xml/service/XmlFileService.java [hunk 3]
```diff
@@ -38,6 +38,12 @@ public XmlFileService(GraphContext ctx)
     public Document loadDocumentQuiet(XmlFileModel model)
     {
         ClassificationService classificationService = new ClassificationService(getGraphContext());
+        if (model.asFile().length() == 0)
+        {
+            LOG.log(Level.WARNING, "Failed to parse xml entity: " + model.getFilePath() + ", as the file is empty.");
```
Comment (lincolnthree) [inline rules-xml/addon/src/main/java/org/jboss/windup/rules/apps/xml/service/XmlFileService.java]: Good.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/58/
Test PASSed.

**Pull request merged**